R/net.dat.R

```diff
@@ -0,0 +1,385 @@
+
+  if (is.null(indexes)) {
+    out <- dat[["epi"]][[item]]
+
+  } else {
+    if (is.logical(indexes)) {
+      if (length(indexes) != dat$control$nsteps)
+        stop("(logical) `indexes` has to have a length equal to the number of
+              steps planned for for the simulation (control$nsteps)")
+    } else if(is.numeric(indexes)) {
+      if (any(indexes > dat$control$nsteps))
+        stop("Some (numeric) `indexes` are larger than the number of
+              steps planned for for the simulation (control$nsteps)")
+    } else {
+      stop("`indexes` must be logical, numeric, or NULL")
+    }
+
+    out <- dat[["epi"]][[item]][indexes]
+  }
+
+  return(out)
+}
+
+#' @rdname dat_get_set
+#' @export
+add_epi <- function(dat, item) {
+  if (item %in% names(dat[["epi"]]))
+    stop(paste("Cannot create the Epidemic outpout, ", item,
+               ": exists already"))
+
+  dat[["epi"]][[item]] <- rep(NA, dat$control$nsteps)
+
+  return(dat)
+}
+
+#' @rdname dat_get_set
+#' @export
+set_epi <- function(dat, item, value) {
+  if (!item %in% names(dat[["epi"]])) {
+    message(paste("Creating Epidemic output: '", item,
+               "'in the Epidemic output list of the Master list object (dat)"))
+
+    dat <- add_epi(dat, item)
+  }
+
+  dat[["epi"]][[item]] <- value
+
+  return(dat)
+}
+
+#' @param at timestep where to add the new value for the Epidemic Outuput `item`
+#' @rdname dat_get_set
+#' @export
+set_epi_at <- function(dat, item, at,  value) {
+  if (length(at) != 1 || !is.numeric(at))
+    stop("`at` must be numeric and of length one")
+
+  if (!item %in% names(dat[["epi"]])) {
+    message(paste("Creating Epidemic output: '", item,
+               "'in the Epidemic output list of the Master list object (dat)"))
+
+    dat <- add_epi(dat, item)
+  }
+
+  if (at > length(dat[["epi"]][[item]])) {
+
+      dat[["epi"]][[item]] <- c(
+        dat[["epi"]][[item]],
+        rep(NA, dat$control$nsteps - length(dat[["epi"]][[item]]))
+      )
+  }
+
+  dat[["epi"]][[item]][at] <- value
+
+  return(dat)
+}
+
+#' @rdname dat_get_set
+#' @export
+get_param_list <- function(dat, item = NULL) {
+  if (is.null(item)) {
+    out <- dat[["param"]]
+
+  } else {
+    missing_item <- setdiff(item,names(dat[["param"]]))
+    if (length(missing_item) > 0)
+      stop(paste("There is no parameters called",
+                 paste(missing_item, collapse = ", "),
+                 "in the parameter list of the Master list object (dat)"))
+
+    out <- dat[["param"]][item]
+  }
+
+  return(out)
+}
+
+#' @rdname dat_get_set
+#' @export
+get_param <- function(dat, item) {
+  if (!item %in% names(dat[["param"]]))
+      stop(paste("There is no parameter called", item,
+                 "in the parameter list of the Master list object (dat)"))
+
+  out <- dat[["param"]][[item]]
+
+  return(out)
+}
+
+#' @rdname dat_get_set
+#' @export
+add_param <- function(dat, item) {
+  if (item %in% names(dat[["param"]]))
```

Comment:
I'm not sure if we would ever want to add/set parameters within modules. These are always just pulled from the param list. Can you think of a reason?

Reply:
I supposed you wanted to use this kind of syntax for all interaction with the `dat` object (i.e. also in the `param` functions. Can remove otherwise.

Reply:
No problem leaving them in there, for consistency at the least. Maybe some users will have a need.